c10/core/TensorImpl.h

```diff
@@ -734,15 +737,14 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
    */
   void set_sizes_contiguous(IntArrayRef new_size) {
     TORCH_CHECK(allow_tensor_metadata_change(), "set_sizes_contiguous is not allowed on Tensor created from .data or .detach()");
-    auto old_dim = sizes_.size();
     auto new_dim = new_size.size();
 
     sizes_.resize(new_dim);
     for (size_t dim = 0; dim < new_dim; ++dim) {
       sizes_[dim] = new_size[dim];
     }
 
-    update_to_contiguous_strides(old_dim);
```

Comment:
Huh. I'm surprised how this ever worked.

Reply:
More or less because `set_sizes_contiguous` called on empty tensor or just resized tensor, this is why huge WARNING in the comment.